File: common/src/main/java/net/eve0415/spigot/WebsocketIntegration/Util/WSIBootstrap.java
package net.eve0415.spigot.WebsocketIntegration.Util;

import net.eve0415.spigot.WebsocketIntegration.WebsocketManager;

public interface WSIBootstrap {
    void onEnable();

    void onDisable();

    WSIPlatformType getPlatformType();

    WSIConfiguration getWSIConfig();

    WSILogger getWSILogger();

    WebsocketManager getWebsocketManager();
}

File: common/src/main/kotlin/net/eve0415/mc/WebsocketIntegration/Util/WSIBootstrap.kt
package net.eve0415.spigot.WebsocketIntegration.Util

import net.eve0415.spigot.WebsocketIntegration.WebsocketManager

public interface WSIBootstrap {
  val platformType: WSIPlatformType

  var serverID: Int

  val config: WSIConfiguration

  val logger: WSILogger

  val websocketManager: WebsocketManager

  fun onEnable()

  fun onDisable()

  fun getServerPort(): Int?

  fun handleChatMessage(name: String, uuid: String, url: String, message: String)

  fun sendServerInfo()
}
